packages/files/README.md

```diff
@@ -0,0 +1,251 @@
+- `signature` is an HS256 JWT (expires in 5 minutes) signed with `clients[fileClientId].secret`. Payload contains file meta, including `recordId`, original filename, mimetype, encoding, and `meta` with app/user info.
+
+### FileRecord model and signature payload
+
+The signed payload corresponds to the public meta of `FileRecord`:
+
+```json
+{
+  "id": "<storage-id>",
+  "recordId": "<FileRecord.id>",
+  "path": "<string|null>",
+  "filename": "<generated-name>",
+  "originalFilename": "<original name>",
+  "mimetype": "<mime>",
+  "encoding": "<encoding>",
+  "meta": {
+    "dv": 1,
+    "sender": { "dv": 1, "fingerprint": "<string>" },
+    "user": "<uuid>",
+    "fileClientId": "<string>",
+    "modelNames": ["<string>", "..."],
+    "sourcefileClientId": "<string|null>"
+  },
+  "iat": <number>,
+  "exp": <number>
+}
+```
+
+### Connect a file to a GraphQL model
+
+After you receive a file `signature` from the upload/share endpoint, pass it into a mutation for a model that has a file field. The field expects an input of type `FileMeta` with a `signature` property.
+
+- Requirements:
+  - The current user must be the owner indicated by `user`.
+  - The model’s list key must be present in `meta.modelNames`.
+  - Use the signature before it expires.
+
+Create example:
+
+```graphql
+mutation CreateSomeModel($data: SomeModelCreateInput!) {
+  obj: createSomeModel(data: $data) {
+    file { filename mimetype publicUrl path }
+  }
+}
+
+// Data example:
+data: {
+  ...otherModelFields,
+  file: { signature: 'signed-string-received-from-file-server' }
+}
+
+```
+
+
+On read, the field resolves to:
+
+```graphql
+{
+  file {
+    id
+    path
+    filename
+    originalFilename
+    mimetype
+    encoding
+    publicUrl
+    meta
+  }
+}
+```
+
+### Sharing a file
+
+Creates a new file record that points to the original binary; useful to “reassign” an uploaded file to another app/user/model.
+
+- Method: `POST`
+- Content-Type: `application/json`
+- Body:
+  - `dv`: `1`
+  - `sender`: `{ dv: 1, fingerprint: string }`
+  - `id`: UUID of existing file record (must belong to the authenticated user)
+  - `user`: `{ id: "uuid" }` UUID of the target user (new owner)
+  - `fileClientId`: target app id (must exist in `clients`)
+  - `modelNames` (optional): array of strings
+
+Example:
+
+```bash
+curl -X POST "$SERVER/api/files/share" \
+  -H "Cookie: $COOKIE" -H "Content-Type: application/json" \
+  -d '{
+    "dv":1,
+    "sender":{"dv":1,"fingerprint":"test-runner"},
+    "id":"<EXISTING_FILE_ID>",
+    "user":{"id": "<TARGET_USER_ID>"},
+    "fileClientId":"another-app",
+    "modelNames":["AnotherModel"]
+  }'
+```
+
+Successful response:
+
+```json
+{
+  "data": {
+    "file": { "id": "<uuid>", "signature": "<jwt>" }
+  }
+}
+```
+
+- The returned `signature` is signed with the target `fileClientId` secret and includes `meta.sourcefileClientId` (original app) and `recordId` of the new file record.
+
+### Attaching a file to a model
```

Comment:
It's like a "Connect a file to a GraphQL model" above but rest way?
Maybe it's worth combining them in one part?